lib/t/feeds.ex

```diff
@@ -376,49 +353,100 @@ defmodule T.Feeds do
   end
 
   # TODO test pubsub
-  def maybe_activate_session(
-        user_id,
-        gender,
-        feed_filter \\ %FeedFilter{genders: ["F", "M", "N"]}
-      ) do
+  @spec maybe_activate_session(Ecto.UUID.t(), String.t(), %FeedFilter{}) :: %LiveSession{}
+  def maybe_activate_session(user_id, gender, feed_filter) do
     is_already_live? = LiveSession |> where(user_id: ^user_id) |> Repo.exists?()
 
-    case is_already_live? do
-      false ->
-        name = Profile |> where(user_id: ^user_id) |> select([p], p.name) |> Repo.one!()
-        m = "user #{name} (#{user_id}) activated LIVE session"
-        Logger.warn(m)
-        Bot.async_post_message(m)
+    unless is_already_live? do
+      profile = FeedProfile |> where(user_id: ^user_id) |> Repo.one!()
+      m = "user #{profile.name} (#{user_id}) activated LIVE session"
+      Logger.warn(m)
+      Bot.async_post_message(m)
 
-        live_session =
-          %LiveSession{user_id: user_id}
-          |> Repo.insert!(on_conflict: :replace_all, conflict_target: :user_id)
-          |> notify_subscribers(gender, feed_filter, :live)
+      live_session =
+        Repo.insert!(%LiveSession{user_id: user_id},
+          on_conflict: :replace_all,
+          conflict_target: :user_id
+        )
 
-        maybe_schedule_push_to_matches(user_id)
-        live_session
+      unless profile.hidden? do
+        blockers =
+          T.Accounts.UserReport
+          |> where([r], r.on_user_id == ^user_id or r.from_user_id == ^user_id)
+          |> select([r], [r.on_user_id, r.from_user_id])
+          |> Repo.all()
+          |> Enum.map(fn users -> users -- [user_id] end)
+
+        matches =
+          Match
+          |> where([m], m.user_id_1 == ^user_id or m.user_id_2 == ^user_id)
+          |> select([m], {[m.user_id_1, m.user_id_2], m.id})
+          |> Repo.all()
+          |> Enum.map(fn {mates, match_id} ->
+            [mate_id] = mates -- [user_id]
+            {mate_id, match_id}
+          end)
+
+        matches
+        |> Enum.map(fn {mate_id, _match_id} ->
+          DispatchJob.new(%{
+            "type" => "match_went_live",
+            "user_id" => user_id,
+            "for_user_id" => mate_id
+          })
+        end)
+        |> Oban.insert_all()
+
+        broadcast_new_live_session(
+          profile,
+          gender,
+          feed_filter.genders,
+          blockers,
+          Map.new(matches)
+        )
+      end
 
-      true ->
-        nil
+      live_session
     end
   end
 
-  defp maybe_schedule_push_to_matches(user_id) do
-    Match
-    |> where([m], m.user_id_1 == ^user_id or m.user_id_2 == ^user_id)
-    |> select([m], {m.user_id_1, m.user_id_2})
-    |> Repo.all()
-    |> Enum.map(fn {user_id_1, user_id_2} ->
-      [mate_id] = [user_id_1, user_id_2] -- [user_id]
+  defp gendered_live_topic, do: "__live"
```

Comment:
As this is not a "gendered" topic anymore, I'll rename it to `live_topic` and review the PR one more time and set the "merger" to be you, @akhrail1996, so that you'd merge it when appropriate.